community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StoreSingleNodeCursor.java

```diff
@@ -192,7 +192,7 @@ public boolean hasLabel( int labelId )
     @Override
     public boolean isDense()
     {
-        return nodeRecord.isDense();
+        return state != null && state.nodeIsAddedInThisTx( nodeId )  ? false : nodeRecord.isDense();
```

Comment:
Is a record added in this transaction never dense? Or am I not reading this correctly?